zipline/finance/blotter.py

```diff
@@ -329,20 +329,23 @@ def get_transactions(self, bar_data):
                     if not order.open:
                         closed_orders.append(order)
 
+        self.prune_orders(closed_orders)
+
+        # FIXME this API doesn't feel right (returning two things here)
+        return transactions, None
+
+    def prune_orders(self, closed_orders):
         # remove all closed orders from our open_orders dict
         for order in closed_orders:
             sid = order.sid
             try:
                 sid_orders = self.open_orders[sid]
                 sid_orders.remove(order)
-            except KeyError:
+            except ValueError:
```

Comment:
When do we expect this case?  Also, could we move the slice statement out of the try, so it's clearer that that isn't generating the error?